task-01/src/App.js

```diff
@@ -0,0 +1,125 @@
+import React, { useState, useEffect, useReducer } from 'react';
+import './App.css';
+
+function reducer(state, action){
+  switch (action.type) {
+    case "inputChange": {
+      const newValue = action.payload;
+      return newValue;
+    }
+    default: {
+  return {...state};
+  }
+  }
+  
+}
+
+function App() {
+  const [personData, setPersonData] = useState({
```

Comment:
it's a general suggestion to keep state as flat as possible. UseReducer is the hook that should be used in case we have a complex state.